commit acfb4d7b4acfb54839d252c094f1266d31ec7ded (refs/changes/91/3591/3)
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2018-03-13T15:19:28-07:00 (1 year, 7 months ago)
    
    TRITON-53 x-account image clone
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/CHANGES.md b/CHANGES.md
index fa5e924..6e7e314 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -8,6 +8,10 @@
 
 ## not yet released
 
+## 10.6.0
+
+- TRITON-53 x-account image clone. Adds the imgapi.cloneImage method.
+
 ## 10.5.0
 
 - DOCKER-524 Implement docker push
diff --git a/lib/imgapi.js b/lib/imgapi.js
index 0431eed..94dee9e 100644
--- a/lib/imgapi.js
+++ b/lib/imgapi.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright 2017 Joyent, Inc.
+ * Copyright 2018 Joyent, Inc.
  */
 
 /*
@@ -2459,6 +2459,56 @@ function updateImage(uuid, data, account, options, callback) {
 };
 
 
+/**
+ * Clones an image.
+ *
+ * @param {String} uuid : the UUID of the image.
+ * @param {UUID} account : The UUID of the account on behalf of whom
+ *      this request is being made. This UUID must be on to the image ACL.
+ * @param options {Object} Optional request options.
+ *      - headers {Object} Optional extra request headers.
+ * @param {Function} callback : `function (err, image, res)`
+ */
+IMGAPI.prototype.cloneImage =
+function cloneImage(uuid, account, options, callback) {
+    var self = this;
+    assert.string(uuid, 'uuid');
+    assert.string(account, 'account');
+    if (typeof (options) === 'function') {
+        callback = options;
+        options = {};
+    }
+    assert.object(options, 'options');
+    assert.optionalObject(options.headers, 'options.headers');
+    assert.func(callback, 'callback');
+
+    var path = self._path(format('/images/%s/clone', uuid), {
+        account: account,
+        channel: self.channel
+    });
+    self._getAuthHeaders(function (hErr, headers) {
+        if (hErr) {
+            callback(hErr);
+            return;
+        }
+        if (options.headers) {
+            simpleMerge(headers, options.headers);
+        }
+        var reqOpts = {
+            path: path,
+            headers: headers
+        };
+        self.client.post(reqOpts, function (err, req, res, image) {
+            if (err) {
+                callback(err, null, res);
+            } else {
+                callback(null, image, res);
+            }
+        });
+    });
+};
+
+
 /**
  * Delete an image.
  *
diff --git a/package.json b/package.json
index a319985..48cf29e 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "sdc-clients",
   "description": "node.js client libraries for Triton core REST APIs.",
-  "version": "10.5.0",
+  "version": "10.6.0",
   "homepage": "http://www.joyent.com",
   "repository": {
     "type": "git",
